vmdb/spec/requests/api/collections_spec.rb

```diff
@@ -0,0 +1,186 @@
+#
+require 'spec_helper'
+
+describe ApiController do
+
+  include Rack::Test::Methods
+
+  let(:api_user)   { @cfme[:user] }
+  let(:zone)       { FactoryGirl.create(:zone, :name => "api_zone") }
+  let(:miq_server) { FactoryGirl.create(:miq_server, :zone => zone) }
+  let(:template) do
+    FactoryGirl.create(:miq_template, :name => "template 1", :vendor => "vmware", :location => "template1.vmtx")
+  end
+
+  before(:each) do
+    init_api_spec_env
+  end
+
+  def app
+    Vmdb::Application
+  end
+
+  def test_collection_query(collection, collection_url, klass, attr = :id)
+    api_basic_authorize
+
+    run_get "#{collection_url}?expand=resources"
+
+    expect_query_result(collection, klass.count, klass.count)
+    expect_result_resources_to_include_data("resources", attr.to_s => klass.pluck(attr))
+  end
+
+  context "Collections" do
+
+    it "query Automation Requests" do
+      FactoryGirl.create(:automation_request)
+      test_collection_query(:automation_requests, automation_requests_url, AutomationRequest)
+    end
+
+    it "query Availability Zones" do
+      FactoryGirl.create(:availability_zone)
+      test_collection_query(:availability_zones, availability_zones_url, AvailabilityZone)
+    end
+
+    it "query Clusters" do
+      FactoryGirl.create(:ems_cluster)
+      test_collection_query(:clusters, clusters_url, EmsCluster)
+    end
+
+    it "query Conditions" do
+      FactoryGirl.create(:condition)
+      test_collection_query(:conditions, conditions_url, Condition)
+    end
+
+    it "query Data Stores" do
+      FactoryGirl.create(:storage)
+      test_collection_query(:data_stores, data_stores_url, Storage)
+    end
+
+    it "query Events" do
+      FactoryGirl.create(:miq_event)
+      test_collection_query(:events, events_url, MiqEvent)
+    end
+
+    it "query Flavors" do
+      FactoryGirl.create(:flavor)
+      test_collection_query(:flavors, flavors_url, Flavor)
+    end
+
+    it "query Groups" do
+      FactoryGirl.create(:miq_group)
+      test_collection_query(:groups, groups_url, MiqGroup)
+    end
+
+    it "query Hosts" do
+      FactoryGirl.create(:host)
+      test_collection_query(:hosts, hosts_url, Host, :guid)
+    end
+
+    it "query Policies" do
+      FactoryGirl.create(:miq_policy)
+      test_collection_query(:policies, policies_url, MiqPolicy)
+    end
+
+    it "query Policy Actions" do
+      FactoryGirl.create(:miq_action)
+      test_collection_query(:policy_actions, policy_actions_url, MiqAction)
+    end
+
+    it "query Policy Profiles" do
+      FactoryGirl.create(:miq_policy_set)
+      test_collection_query(:policy_profiles, policy_profiles_url, MiqPolicySet)
+    end
+
+    it "query Providers" do
+      FactoryGirl.create(:ext_management_system)
+      test_collection_query(:providers, providers_url, ExtManagementSystem, :guid)
+    end
+
+    it "query Provision Requests" do
+      FactoryGirl.create(:miq_provision_request, :source => template, :userid => api_user)
```

Comment:
just the base :miq_provision_request in factories/miq_provision_request.rb with no parameters.